app/src/main/java/com/example/android/sunshine/app/ForecastAdapter.java

```diff
@@ -81,60 +43,46 @@ public ForecastAdapter(Context context, OnClickHandler onClickHandler, boolean s
 
     @Override
     public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-        // Choose the layout type
-        int layoutId = -1;
-        switch (viewType) {
-            case VIEW_TYPE_TODAY: {
-                layoutId = R.layout.list_item_forecast_today;
-                break;
-            }
-            case VIEW_TYPE_FUTURE_DAY: {
-                layoutId = R.layout.list_item_forecast;
-                break;
-            }
-        }
-
-        View view = LayoutInflater.from(parent.getContext()).inflate(layoutId, parent, false);
+        View view = LayoutInflater.from(parent.getContext()).inflate(viewType, parent, false);
         view.setFocusable(true);
         return new ViewHolder(view, selector, onClickHandler);
     }
 
     @Override
     public void onBindViewHolder(ViewHolder holder, int position) {
-        cursor.moveToPosition(position);
+        WeatherConditions conditions = weatherConditionsList.get(position);
 
         int viewType = holder.getItemViewType();
-        int weatherId = cursor.getInt(WeatherContract.COL_WEATHER_CONDITION_ID);
+        int weatherId = conditions.weatherId();
         int fallbackIconId;
         switch (viewType) {
             case VIEW_TYPE_TODAY: {
                 // Get weather icon
-                fallbackIconId = Utility.getArtResourceForWeatherCondition(weatherId);
+                fallbackIconId = WeatherUtility.getArtResourceForWeatherCondition(weatherId);
```

Comment:
I would like to keep `WeatherConditions` a simple data class, by moving the behaviour from the Utility class it would get very cluttered, IMO.